### ISSUE TITLE
Combination of MethodImpl and ref parameters confused IDE0060 rule

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn-analyzers#6228]
### Analyzer

**Diagnostic ID**: [IDE0060](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0060): `Remove unused parameter`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 

```
    <MicrosoftCodeAnalysisAnalyzersVersion>3.3.4-beta1.22524.2</MicrosoftCodeAnalysisAnalyzersVersion>
    <MicrosoftCodeAnalysisNetAnalyzersVersion>7.0.0-preview1.22524.2</MicrosoftCodeAnalysisNetAnalyzersVersion>
```

### Describe the bug

IDE0060 warning is reported incorrectly

### Steps To Reproduce

```c#
        [MethodImpl(MethodImplOptions.NoInlining)]
        private static void Test(ref byte destination, ref byte source)
        {
            destination = ref Unsafe.AddByteOffset(ref destination, 1);
            source = ref Unsafe.AddByteOffset(ref source, 1);
        }
```

### Expected behavior

No warning

### Actual behavior

```
error IDE0060: Parameter 'destination' can be removed; its initial value is never used 
error IDE0060: Parameter 'source' can be removed; its initial value is never used 
```

## Additional context

It's strange that removing `MethodImplAttribute` fixes the problem

**Comment by MichaelSimons** [comment]
I just encountered this while source-building aspnetcore in .NET 8.0.  This [method](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/src/Matching/Ascii.cs#L15) produces the following errors:

```
    /repos/tarball-main/src/aspnetcore/artifacts/source-build/self/src/src/Http/Routing/src/Matching/Ascii.cs(16,94): error IDE0060: Parameter 'length' can be removed; its initial value is never used [/repos/tarball-main/src/aspnetcore/artifacts/source-build/self/src/src/Http/Routing/src/Microsoft.AspNetCore.Routing.csproj]
    /repos/tarball-main/src/aspnetcore/artifacts/source-build/self/src/src/Http/Routing/src/Matching/Ascii.cs(16,87): error IDE0060: Parameter 'b' can be removed; its initial value is never used [/repos/tarball-main/src/aspnetcore/artifacts/source-build/self/src/src/Http/Routing/src/Microsoft.AspNetCore.Routing.csproj]
    /repos/tarball-main/src/aspnetcore/artifacts/source-build/self/src/src/Http/Routing/src/Matching/Ascii.cs(16,65): error IDE0060: Parameter 'a' can be removed; its initial value is never used [/repos/tarball-main/src/aspnetcore/artifacts/source-build/self/src/src/Http/Routing/src/Microsoft.AspNetCore.Routing.csproj]
```

Removing the `[MethodImpl(MethodImplOptions.AggressiveInlining)]` gets around the build issue.

**Comment by MichaelSimons** [comment]
cc @333fred - Can you help direct this?  Is this a Roslyn issue?
